src/control/tools/BaseStrokeHandler.cpp

```diff
@@ -15,9 +15,10 @@ guint32 BaseStrokeHandler::lastStrokeTime;  // persist for next stroke
 
 BaseStrokeHandler::BaseStrokeHandler(XournalView* xournal, XojPageView* redrawable, const PageRef& page, bool flipShift,
                                      bool flipControl):
-        InputHandler(xournal, redrawable, page), SnapToGridInputHandler(xournal->getControl()->getSettings()) {
+        InputHandler(xournal, redrawable, page) {
     this->flipShift = flipShift;
     this->flipControl = flipControl;
+    snappingHandler = new SnapToGridInputHandler(xournal->getControl()->getSettings());
 }
```

Comment:
Do not use new / delete in newly written c++ code anymore.
Use modern member initialization:

```suggestion
        InputHandler(xournal, redrawable, page), snappingHandler(xournal->getControl()->getSettings()), flipShift(flipShift), flipControl(flipControl) {}
```